zcash_client_backend/src/wallet.rs

```diff
@@ -259,21 +262,21 @@ impl WalletNote {
 /// Information about a note that is tracked by the wallet that is available for spending,
 /// with sufficient information for use in note selection.
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub struct ReceivedNote<NoteRef> {
+pub struct ReceivedNote<NoteRef, NoteT> {
     note_id: NoteRef,
     txid: TxId,
     output_index: u16,
-    note: WalletNote,
+    note: NoteT,
```

Comment:
I agree that `WalletNote` could usefully be renamed `Note`, but disagree with your other comment: I frequently use polymorphic data types to "smuggle" additional data, and then `map` them back to the form expected by various call sites. It's entirely possible that it could usefully be something other than a note here.